### ISSUE TITLE
Loading the weights - Bias term

### ISSUE THREAD

**Issue by victormelo** [luizgh/sigver_wiwd#1]
Hello,

I am trying to load the model weights you made available, but it doesn't seem that I can access the bias term of the Convolutional Layers.

When I run:
```
[l.shape for l in lasagne.layers.get_all_param_values(model.model['fc2'])]
```
Those are the params I get:
```
[W, beta, gamma, mean, inv_std, W, beta, gamma, mean, inv_std, W, beta, gamma, mean, inv_std, W, beta, gamma, mean, inv_std, W, beta, gamma, mean, inv_std, W, beta, gamma, mean, inv_std, W, beta, gamma, mean, inv_std]
```

and 
```
[l.shape for l in lasagne.layers.get_all_param_values(model.model['fc2'])]
      
```

I get:
```
[(96, 1, 11, 11), (96,), (96,), (96,), (96,), (256, 96, 5, 5), (256,), (256,), (256,), (256,), (384, 256, 3, 3), (384,), (384,), (384,), (384,), (384, 384, 3, 3), (384,), (384,), (384,), (384,), (256, 384, 3, 3), (256,), (256,), (256,), (256,), (3840, 2048), (2048,), (2048,), (2048,), (2048,), (2048, 2048), (2048,), (2048,), (2048,), (2048,)]
```

As we can see, those are the params related to the weights of the convolutional layers and the params of the batch normalization, it doesnt seem that this method returns the bias term as one of the params.

Any idea how can I get the bias information?

I am asking this because I want to port this model to Tensorflow.

Best regards,
Victor.

**Comment by luizgh** [comment]
Hi Victor,

We don't use biases on the convolutions when using batch normalization, since the biases would be removed during batch normalization (when the mean is subtracted). Instead, the biases are added as part of the batch normalization layer, but they are called "beta" in this context.

For your reference, I have added a tensorflow implementation of the model, that uses the same weights. See the file "tf_example.py". Note that I only implemented the "test" version of batch normalization. If you want re-train / finetune the model you would need to adapt the code (use the tf implementation of batch normalization, but you need to check the details. e.g. Lasagne stores the inverse of the standard deviation, which may not be the case for tensorflow (it may store the standard deviation itself)).

Let me know if that solves your problem.

**Comment by victormelo** [comment]
Hello Luiz,

Thank you very much for your attention and kindness in providing me a working test version for Tensorflow.

Best regards,
Victor.
